app/controllers/auth0_controller.rb

```diff
@@ -0,0 +1,83 @@
+class Auth0Controller < ApplicationController
```

Comment:
The `Auth0Controller` is directly concatenating query parameters to construct the authorization URL, which can lead to URL manipulation vulnerabilities. Consider using built-in Rails helpers like `url_for` or `URI` methods to safely construct query parameters.